libs/gl-client/src/lnurl/tor_http_client.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::Result;
+use arti_client::{TorClient, TorClientConfig};
+use log::info;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use url::Url;
+
+pub struct TorHttpClient {
+    tor_client: TorClient<tor_rtcompat::PreferredRuntime>,
+}
+
+impl TorHttpClient {
+    pub async fn new() -> Result<Self> {
+        let mut config_builder = TorClientConfig::builder();
+        let address_filter = config_builder.address_filter();
+        address_filter.allow_onion_addrs(true);
+        address_filter.build()?;
```

Comment:
The `address_filter` is not used anywhere? And the fact that `address_filter` is retrieved from `config_builder` but not `mut` suggests that any change to it will not be reflected in `config_builder`. Or the API might just be strange :-)

Reply:
`address_filter` does not need to be marked as `mut` to contain a mutable reference.

See:
```
let mut a = A {x: 2};
// b is analogous to address_filter in the referenced code snippet (address_filter is of type '&mut ClientAddrConfigBuilder')
let b = &mut a;
b.x = 4;
println!("{}", b.x);
```